bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -756,10 +889,47 @@ pub struct Session {
     pub device_id: String,
 
     // FFI-only fields (for now)
+    /// The URL for the homeserver used for this session.
     pub homeserver_url: String,
+    /// Additional data for this session session if OpenID Connect was used
```

Comment:
nit: session session

Also, this is a string that contains a serialized JSON blob, breaking the FFI layer boundary by allowing stringly typed data.

A few questions (very uninformed, as I don't know much about OIDC at all):

- does it _need_ to be stored, or could it be transient and passed as a function argument somewhere?
- if it does need to be stored, could it be stored as the final deserialized format instead of JSON?

Reply:
On the stringly typed data and your questions:
- Yes it needs to be stored, it contains additional OIDC related data that is required for the app to restore a session correctly, so storing it alongside everything else makes the most sense to me.
- The app has absolutely no interest in the contents of this struct. If we make it a `struct`, we would need to a) convert all these [properties](https://github.com/matrix-org/matrix-authentication-service/blob/ff1f33d94a5ab37afea01be8d7f8b7c6907175f1/crates/oauth2-types/src/registration/mod.rs#L131-L439) to types that Uniffi is happy to bridge and b) Then implement Codable conformances on the Swift side (and presumably similar on the Kotlin side) which seems like a lot of unnecessary work given we aren't interested in the data.